interface/orders/gen_hl7_order.inc.php

```diff
@@ -141,7 +141,7 @@ function loadPayerInfo($pid, $date = '')
     $payers = array();
     $dres = sqlStatement(
         "SELECT * FROM insurance_data WHERE " .
-        "pid = ? AND date <= ? ORDER BY type ASC, date DESC",
+        "pid = ? AND (date <= ? OR date IS NULL) AND provider != '' AND provider IS NOT NULL ORDER BY type ASC, date DESC",
```

Comment:
actually, on further thought, none of the provider stuff is needed since null used to 0000-00-00 which were all included in all of these queries

Reply:
nice!